src/email_requests/views.py

```diff
@@ -65,8 +66,17 @@ class ListOwnRequestsView(generics.ListAPIView):
     serializer_class = RequestSerializer
 
     def get_queryset(self):
-        mentor = get_object_or_404(Mentor, profile__user=self.request.user)
-        return Request.objects.filter(mentor=mentor).order_by('date_created').reverse()
+        profile = get_object_or_404(Profile, user=self.request.user)
+
+        try:
+            mentor = Mentor.objects.get(profile=profile)
+        except Mentor.DoesNotExist:
```

Comment:
Please avoid try and except.

Use mentor = Mentor.objects.filter(profile=profile).first() which will do this for you

Reply:
I'll change the rest of these